### ISSUE TITLE
manage.present/manage.alived not working

### ISSUE THREAD

**Issue by harmv** [saltstack/salt#39524]
### Description of Issue/Question

I noticed a regression when upgrading from 2016.3.3 to 2016.11.2

manage.alived and friends return incorrect data. They report _every_ minion the master has seen in the past as 'present'

### Setup

I have a single minion `14100018` connected to a remote master via `transport=tcp`

### Steps to Reproduce Issue

1. Start the minion 14100018,   `systemctl start salt-minion`
2. Stop the minion 14100018,  `systemctl stop salt-minion`
3. on master run: `$ salt manage.present`  or `$ salt manage.alived`

### Expected result
I expect minion `14100018`  _not_ to be listed as present

### Actual result
```
# salt-run manage.present
- 12345
- 14100018
```

The presence events are handled ok, when listening to salt events _I_ do see the correct ones passing.

```
# python listen_salt_events.py 

salt/auth
salt/presence/change
new: ['14100018']
lost: []
salt/presence/present
present: ['14100018']
minion_start
salt/minion/14100018/start
salt/presence/change
new: []
lost: ['14100018']
salt/presence/present
present: []

Aborted with CTRL-C
```

Above you see that 14100018 came up, and than it was shut down.  the `present` field of the final present is an empty list. `[]`
Still when running manage.present... it lists our minion....
```
# salt-run manage.present
- 12345                 <-----   more WTF, that test minion ran once... long time ago. thats long gone 
- 14100018            <---- WTF, its shutdown.
```

### Versions Report

salt master: 2016.11.2
salt minion: 2016.11.2

master
```
transport: tcp
presence_events: True
```
not connected via localhost. (minion `14100018` is remote)

2016.3.3 worked better.

Broken also on 2016.11.2 when using transport=zmq instead of transport=tcp.

**Comment by Ch3LL** [comment]
@harmv I have not been able to replicate this behavior. I know there is an issue related to this but that is when the minion is on the same host as the master and you stated these are remote connections so that must not be related. You stated you saw this behavior when you upgraded, what if you remove the cache on the master side? Also can you check what kind of network data is being sent and if the minion is seeing that data at all? 

Also can you make sure when the minion is stopped to make sure there are no other defunct minion processes. Thanks

**Comment by harmv** [comment]
Ch3LL wrote:
>You stated you saw this behavior when you upgraded, what if you remove the cache on the master side?

How can I clear the cache ?

I tried this on the master, but that somehow crashes ?

```
# salt-run cache.clear_all
Exception occurred in runner cache.clear_all: Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/client/mixins.py", line 397, in _low
    if 'data' in data['return']:
TypeError: argument of type 'bool' is not iterable
```

Can I manually clear te cache by removing some files on the master?

Ch3LL wrote:
>Also can you make sure when the minion is stopped to make sure there are no other defunct minion processes. Thanks

I'm sure about that, besides shutting down the minion ('14100018'), I tested with shutting down the machine the minion runs on.
8 hours later the salt master keeps listing it in:

```
# salt-run manage.present
- 12345
- 14100018
```


`manage.up`  reports (correctly) that all minions are down.

```
root@salt2:~# salt-run manage.up
root@salt2:~#
```

**Comment by Ch3LL** [comment]
You can manually remove by doing the following:

1. Stop salt master
2. `rm -r /var/cache/salt/master`
3. Start salt master

**Comment by harmv** [comment]
Clearing the cache has no effect.

I cleared the cache
```
# I first stopped the remote minion
root@salt2:~# systemctl stop salt-master
root@salt2:~# rm -r /var/cache/salt/master
root@salt2:~# systemctl start salt-master
root@salt2:~# salt-run manage.alived
root@salt2:~# salt-run manage.up
root@salt2:~#
```
So far so good, no minions are reported.... for now clearing the cache helped.
however... read on.

Then I fire up the minion (`systemctl start salt-minion`), and run `manage.` commands on the master.

```
root@salt2:~# salt-run manage.up
- 14100018
root@salt2:~# salt-run manage.alived
- 14100018
```
This is still correct, both the online check (`manage.up`) as the offline check (`manage.alived`)  report that our minion is online

Then I stop the minion (`systemctl stop salt-minion`), and check the `manage.` commands again on the master

```
root@salt2:~# salt-run manage.up
root@salt2:~# salt-run manage.alived
- 14100018
```
This is **incorrect**.  The online check correctly sees that the minion is down.  But the offline check  insist that the minion is online.

This stays online.  restarting  the master, shutting down the minion machine. It never changes, It keeps reported that 14100018 is online.


## master config

transport tcp, we _do_ use master_job cache.. But that should be unrelated, no?
`/etc/salt/master/`
```
interface: 0.0.0.0
transport: tcp
presence_events: True
master_job_cache: postgres_local_cache
master_job_cache.postgres.host: 'localhost'
master_job_cache.postgres.user: '******'
master_job_cache.postgres.passwd: '******'
master_job_cache.postgres.db: '*****'
master_job_cache.postgres.port: 5432
```

**Comment by harmv** [comment]
Same with 2016.11.3 (salt-master)

**Comment by Ch3LL** [comment]
I believe you are running into this issue at this point: https://github.com/saltstack/salt/issues/33466

**Comment by harmv** [comment]
Yes that looks like the same problem

(and the fix for that problem PR#36126 is indeed not in my version: 2016.11.3)

Note: That pull requests mentions that it only fixes the zmq case, not the transport=tcp case.

Is more work needed for that ticket?

**Comment by harmv** [comment]
#33466 got reopened.
This ticket can be closed as duplicate of  #33466
